libenv/sysinfo.c

```diff
@@ -1645,6 +1659,95 @@ static void OSClasses(EvalContext *ctx)
 
 /*********************************************************************************/
 
+#ifdef __APPLE__
+static void Apple_Version(EvalContext *ctx)
+{
+    FILE *pp = NULL;
+
+    Log(LOG_LEVEL_VERBOSE, "This appears to be an apple system.");
+    Log(LOG_LEVEL_VERBOSE, "Looking for product name and version...");
+    if ((!FileCanOpen("/usr/bin/sw_vers", "r") || ((pp = cf_popen("/usr/bin/sw_vers", "r", true)) == NULL)))
+    {
+        Log(LOG_LEVEL_ERR, "Could not open and run /usr/bin/sw_vers to find macOS system version information.");
+        return;
+    }
+
+    size_t line_size = CF_BUFSIZE;
+    char *line = xmalloc(line_size);
+    int revcomps = 0, major, minor, patch;
```

Comment:
These should be unsigned now
```suggestion
    int revcomps = 0;
    unsigned int major, minor, patch;
```